### DISCUSSION TITLE
Next.js에서 CSR, SSR, SSG, ISR 방식은 언제 사용하면 좋을까요?

### DISCUSSION THREAD

**Discussion by hyjoong** [diving-react/study-react-deepdive#51] [General]
Next.js로 프로젝트를 진행할 때 CSR, SSR, SSG, ISR 방식은 각각 언제 사용할까요?

**Comment by hyjoong** [comment]
공식문서 보니 [On-Demand Revalidation](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration#on-demand-revalidation) 방식도 있네요. 이 방식은 언제 사용하는 걸까요 ..??

**Comment by lledellebell** [comment]
> References:
> - [Next.js 공식 문서 (ISR; Incremental Static Regeneration)](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration)
> - [Vercel 가이드 (ISR; Incremental Static Regeneration)](https://vercel.com/docs/incremental-static-regeneration)
